sdk/servicelinker/azure-resourcemanager-servicelinker/src/samples/java/com/azure/resourcemanager/servicelinker/CreateServiceLinker.java

```diff
@@ -0,0 +1,184 @@
+import com.azure.resourcemanager.AzureResourceManager;
+import com.azure.resourcemanager.appplatform.models.SkuName;
+import com.azure.resourcemanager.appplatform.models.SpringApp;
+import com.azure.resourcemanager.appplatform.models.SpringService;
+import com.azure.resourcemanager.appservice.models.PricingTier;
+import com.azure.resourcemanager.appservice.models.RuntimeStack;
+import com.azure.resourcemanager.appservice.models.WebApp;
+import com.azure.resourcemanager.keyvault.models.Vault;
+import com.azure.resourcemanager.msi.models.Identity;
+import com.azure.resourcemanager.servicelinker.models.AzureResource;
+import com.azure.resourcemanager.servicelinker.models.ClientType;
+import com.azure.resourcemanager.servicelinker.models.LinkerResource;
+import com.azure.resourcemanager.servicelinker.models.SecretAuthInfo;
+import com.azure.resourcemanager.servicelinker.models.SourceConfiguration;
+import com.azure.resourcemanager.servicelinker.models.UserAssignedIdentityAuthInfo;
+import com.azure.resourcemanager.servicelinker.models.ValueSecretInfo;
+import com.azure.resourcemanager.sql.models.SqlDatabase;
+import com.azure.resourcemanager.sql.models.SqlServer;
+import reactor.core.publisher.Mono;
+
+import java.util.Locale;
+import java.util.UUID;
+
+public class CreateServiceLinker {
+    private static final String USER_TOKEN_HEADER = "x-ms-serviceconnector-user-token";
+
+    /**
+     * Main entry point.
+     *
+     * @param args the parameters
+     */
+    public static void main(String[] args) throws Exception {
+        // Authentication environment variables need to be set: AZURE_CLIENT_ID,AZURE_TENANT_ID,AZURE_CLIENT_SECRET,AZURE_SUBSCRIPTION_ID
+        TokenCredential credential = new DefaultAzureCredentialBuilder()
+            .authorityHost(AzureAuthorityHosts.AZURE_PUBLIC_CLOUD)
+            .build();
+
+        AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+
+        // create a policy for setting user token into a specific header for some special scenarios
+        HttpPipelinePolicy userTokenPolicy = (context, next) -> {
+            Mono<String> token = null;
+            String bearerTokenPrefix = "bearer ";
+            String authorization = context.getHttpRequest().getHeaders().getValue("Authorization");
+            if (authorization != null && authorization.toLowerCase(Locale.ROOT).startsWith(bearerTokenPrefix)) {
+                token = Mono.just(authorization.substring(bearerTokenPrefix.length()));
+            } else {
+                token = credential
+                    .getToken(new TokenRequestContext().addScopes(profile.getEnvironment().getResourceManagerEndpoint() + "/.default"))
+                    .map(AccessToken::getToken);
+            }
+
+            return token
+                .flatMap(accessToken -> {
+                    context.getHttpRequest().getHeaders().set(USER_TOKEN_HEADER, accessToken);
+                    return next.process();
+                });
+        };
```

Comment:
If this is typical and recommended, should you define it as a static class so that it be easier for user to reuse?

Reply:
done